### ISSUE TITLE
Model with new layers added changes the result of previous layers

### ISSUE THREAD

**Issue by iumyx2612** [open-mmlab/mmdetection#9051]
### Prerequisite

- [X] I have searched [the existing and past issues](https://github.com/open-mmlab/mmdetection/issues) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmdetection).

### 🐞 Describe the bug

I write a new FCOSHead below: 
```
@HEADS.register_module()
class NewFCOSHead(FCOSHead):
    def __init__(self,
                 **kwargs):
        super(NewFCOSHead, self).__init__(**kwargs)
        self.bbox_coder = RevDistancePointBBoxCoder()

    def _init_predictor(self):
        """Initialize predictor layers of the head."""
        self.conv_cls = nn.Conv2d(
            self.feat_channels, self.cls_out_channels, 3, padding=1)
        self.conv_reg = nn.ModuleList([
            nn.Sequential(
                SELayer(self.feat_channels),
                nn.Conv2d(self.feat_channels, 2, 3, padding=1),
            ),
            nn.Sequential(
                SELayer(self.feat_channels),
                nn.Conv2d(self.feat_channels, 2, 3, padding=1),
            ),
        ])


    def forward_single(self, x, scale, stride):
        """Forward features of a single scale level.

        Args:
            x (Tensor): FPN feature maps of the specified stride.
            scale (:obj: `mmcv.cnn.Scale`): Learnable scale module to resize
                the bbox prediction.
            stride (int): The corresponding stride for feature maps, only
                used to normalize the bbox prediction when self.norm_on_bbox
                is True.

        Returns:
            tuple: scores for each class, bbox predictions and centerness \
                predictions of input feature maps.
        """
        cls_feat = x
        reg_feat = x

        for cls_layer in self.cls_convs:
            cls_feat = cls_layer(cls_feat)
        cls_score = self.conv_cls(cls_feat)

        for reg_layer in self.reg_convs:
            reg_feat = reg_layer(reg_feat)

        lr_pred = self.conv_reg[0](reg_feat)
        tb_pred = self.conv_reg[1](reg_feat)
        bbox_pred = torch.cat([lr_pred, tb_pred],
                              dim=1)
        if self.centerness_on_reg:
            centerness = self.conv_centerness(reg_feat)
        else:
            centerness = self.conv_centerness(cls_feat)
        # scale the bbox_pred of different level
        # float to avoid overflow when enabling FP16
        bbox_pred = scale(bbox_pred).float()
        if self.norm_on_bbox:
            # bbox_pred needed for gradient computation has been modified
            # by F.relu(bbox_pred) when run with PyTorch 1.10. So replace
            # F.relu(bbox_pred) with bbox_pred.clamp(min=0)
            bbox_pred = bbox_pred.clamp(min=0)
            if not self.training:
                bbox_pred *= stride
        else:
            bbox_pred = bbox_pred.exp()
        return cls_score, bbox_pred, centerness
```

This is my config file:
```
_base_ = [
    '../configs/fcos/fcos_center-normbbox-centeronreg-giou_r50_caffe_fpn_gn-head_1x_coco.py'
]

model = dict(
    bbox_head=dict(
        type='NewFCOSHead'
    )
)
```
Script I ran:
`python tools/train.py local_cfg/config.py --seed 0`

When performing a forward pass, I debug the 1st iteration, this is the results from FPN Necks:  

![Screenshot from 2022-10-17 16-08-49](https://user-images.githubusercontent.com/69593462/196137905-a6e94dc2-5c3a-42c8-ac3f-539bf72d57bb.png)

However, when performing a forward pass with **normal FCOS**:
`python tools/train.py configs/fcos/fcos_center-normbbox-centeronreg-giou_r50_caffe_fpn_gn-head_1x_coco.py --seed 0`
The results are different from above:  
![Screenshot from 2022-10-17 16-11-28](https://user-images.githubusercontent.com/69593462/196138537-662f413e-877d-4f4c-a17f-0684c577a001.png)

Feature maps from FPN Necks haven't pass through Bbox_head yet it still different? 

I tried with another Bbox_head below:
```
@HEADS.register_module()
class DecoupledRevFCOSHead_v2(FCOSHead):
    def __init__(self,
                 **kwargs):
        super(DecoupledRevFCOSHead_v2, self).__init__(**kwargs)
        self.bbox_coder = RevDistancePointBBoxCoder()

    def _init_predictor(self):
        """Initialize predictor layers of the head."""
        self.conv_cls = nn.Conv2d(
            self.feat_channels, self.cls_out_channels, 3, padding=1)
        self.conv_reg = nn.ModuleList([
            nn.Conv2d(self.feat_channels, 2, 3, padding=1),
            nn.Conv2d(self.feat_channels, 2, 3, padding=1)
        ])

    def forward_single(self, x, scale, stride):
        """Forward features of a single scale level.

        Args:
            x (Tensor): FPN feature maps of the specified stride.
            scale (:obj: `mmcv.cnn.Scale`): Learnable scale module to resize
                the bbox prediction.
            stride (int): The corresponding stride for feature maps, only
                used to normalize the bbox prediction when self.norm_on_bbox
                is True.

        Returns:
            tuple: scores for each class, bbox predictions and centerness \
                predictions of input feature maps.
        """
        cls_feat = x
        reg_feat = x

        for cls_layer in self.cls_convs:
            cls_feat = cls_layer(cls_feat)
        cls_score = self.conv_cls(cls_feat)

        for reg_layer in self.reg_convs:
            reg_feat = reg_layer(reg_feat)

        lr_pred = self.conv_reg[0](reg_feat)
        tb_pred = self.conv_reg[1](reg_feat)
        bbox_pred = torch.cat([lr_pred, tb_pred],
                              dim=1)
        if self.centerness_on_reg:
            centerness = self.conv_centerness(reg_feat)
        else:
            centerness = self.conv_centerness(cls_feat)
        # scale the bbox_pred of different level
        # float to avoid overflow when enabling FP16
        bbox_pred = scale(bbox_pred).float()
        if self.norm_on_bbox:
            # bbox_pred needed for gradient computation has been modified
            # by F.relu(bbox_pred) when run with PyTorch 1.10. So replace
            # F.relu(bbox_pred) with bbox_pred.clamp(min=0)
            bbox_pred = bbox_pred.clamp(min=0)
            if not self.training:
                bbox_pred *= stride
        else:
            bbox_pred = bbox_pred.exp()
        return cls_score, bbox_pred, centerness
```
When performing a forward pass with this new Bbox_head:

![Screenshot from 2022-10-17 16-19-05](https://user-images.githubusercontent.com/69593462/196139964-6999ab09-c4d4-44bc-bb0b-24a0af703eb3.png)
I get the same results as **normal FCOS**. Why adding a SELayer to `bbox_head` differ the results from **FPN Neck**?

### Environment

sys.platform: linux
Python: 3.9.12 (main, Jun  1 2022, 11:38:51) [GCC 7.5.0]
CUDA available: False
GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
PyTorch: 1.10.0+cpu
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=0, USE_CUDNN=OFF, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.11.0+cpu
OpenCV: 4.5.1
MMCV: 1.5.3
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: not available
MMDetection: 2.25.1+94493b9

Process finished with exit code 0

**Comment by Czm369** [comment]
Maybe you can make sure you input the same image on both sides and then look at the feature map, because the modules initialized on both sides are not the same, even if you fix the random seed, there will be a difference.

**Comment by iumyx2612** [comment]
> Maybe you can make sure you input the same image on both sides and then look at the feature map, because the modules initialized on both sides are not the same, even if you fix the random seed, there will be a difference.

there will be difference even if the seed is the same? Is this a bug or it should be like that?

**Comment by Czm369** [comment]
A fixed random seed can only guarantee that the same code will run the same result multiple times, but it cannot guarantee that the same result will run the code with changes

**Comment by iumyx2612** [comment]
> A fixed random seed can only guarantee that the same code will run the same result multiple times, but it cannot guarantee that the same result will run the code with changes

Thank you!!!
